pkg/lib/cockpit.d.ts

```diff
@@ -206,11 +201,21 @@ declare module 'cockpit' {
     function ngettext(context: string, message1: string, messageN: string, n: number): string;
 
     function format(format_string: string, ...args: unknown[]): string;
-    function format_number(n: number, precision?: number): string
-    function format_bytes(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate?: false }): string;
-    function format_bytes(n: number, factor: 1000 | 1024, options: FormatOptions & { separate: true }): string[];
-    function format_bytes_per_sec(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate?: false }): string;
-    function format_bytes_per_sec(n: number, factor: 1000 | 1024, options: FormatOptions & { separate: true }): string[];
-    function format_bits_per_sec(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate?: false }): string;
-    function format_bits_per_sec(n: number, factor: 1000 | 1024, options: FormatOptions & { separate: true }): string[];
+
+    /* === Number formatting ===================== */
+
+    type FormatOptions = {
+        precision?: number;
+        base2?: boolean;
+    };
+    type MaybeNumber = number | null | undefined;
+
+    function format_number(n: MaybeNumber, precision?: number): string
+    function format_bytes(n: MaybeNumber, options?: FormatOptions): string;
+    function format_bytes_per_sec(n: MaybeNumber, options?: FormatOptions): string;
+    function format_bits_per_sec(n: MaybeNumber, options?: FormatOptions & { base2?: false }): string;
```

Comment:
Just to make sure I understand this: The `base2` restriction is here to make `base2: true` fall through the "deprecated" declaration, right? Not because enabling base2 would actually change the output type.

Reply:
I'm pretty sure that enabling base2 will break this (ie: return just the number) in exactly the same way as passing 1024 as factor would.  That particular function lacks base2 units (ie: no Mibps or so)...

Reply:
Right, and I don't think we ever supported that actually.